src/main/anovos/data_analyzer/association_eval_varclus.py

```diff
@@ -0,0 +1,451 @@
+from pyspark.ml.feature import VectorAssembler, StandardScaler
+from pyspark.mllib.linalg.distributed import RowMatrix, DenseMatrix
+import math
+import collections
+from factor_analyzer import Rotator
+import pandas as pd
+import numpy as np
+import random
+
+
+class VarClusHiSpark(object):
+    """
+    This class is a scalable version of [VarClusHi] [2] library to perform variable clustering on PySpark dataframes
+    with necessary optimizations, and sampling is not required.
+
+    [2]: https://github.com/jingtt/varclushi   "VarCluShi"
+
+    Variable Clustering groups attributes that are as correlated as possible among themselves within a cluster and as
+    uncorrelated as possible with attribute in other clusters. By default, it begins with all variables in a single cluster.
+    It then repeats the following steps:
+    1. A cluster is chosen for splitting. This cluster has the largest eigenvalue associated with the 2nd PC
+    2. Find the first 2 PCs of the chosen cluster and split into 2 clusters, then perform an orthoblique rotation
+    (raw quartimax rotation on the eigenvectors)
+    3. Variables are iteratively re-assigned to clusters to maximize the variance:
+        (a) Nearest Component Sorting (NCS) Phase: In each iteration, the cluster components are computed.
+        Each variable is assigned to the rotated component with which it has the highest squared correlation
+        (b) Search Phase: Assign each variable into different cluster to see if this increase the variance explained. If
+         variable is re-assigned here, the components of the clusters involved are recomputed before next variable is tested
+    The procedure stops when the 2nd eigenvalue of each cluster is smaller than maxeigval2 parameter.
+
+    """
+
+    def __init__(self, df, feat_list=None, maxeigval2=1, maxclus=None, n_rs=0):
+        """
+        Parameters
+        ----------
+        df
+            PySpark Dataframe
+        feat_list
+            List of features to perform variable clustering e.g., ["col1","col2"].
```

Comment:
For all arguments that have a default value, can you put (Default value = ....) instead of e.g.? For example, for this one: (Default value = None). You can refer to here https://github.com/anovos/anovos/blob/e0b3055a33415a211911744efbacafd0f100ad38/src/main/anovos/data_analyzer/association_evaluator.py#L79